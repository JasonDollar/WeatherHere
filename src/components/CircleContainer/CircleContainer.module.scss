.circleContainer {
  & ul {
    display: flex;
    list-style: none;
    justify-content: space-around;
    flex-wrap: wrap;
    padding: 0;

    & li {
      flex: 0 0 50%;
      @media only screen and (min-width: 768px) {
        flex-basis: 25%;
      }
    }

  }
}