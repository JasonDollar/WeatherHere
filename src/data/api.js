const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

