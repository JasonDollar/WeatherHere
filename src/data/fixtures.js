import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}