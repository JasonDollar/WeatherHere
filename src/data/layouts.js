const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



