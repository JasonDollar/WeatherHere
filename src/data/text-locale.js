const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 