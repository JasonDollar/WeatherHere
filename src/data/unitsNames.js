export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}