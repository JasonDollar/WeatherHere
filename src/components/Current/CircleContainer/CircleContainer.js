import React from 'react'
import DataCircle from './DataCircle/DataCircle'
import { formatNumber } from '../../../data/utils'

import classes from './CircleContainer.module.scss'


const CircleContainer = ({ currently, currentText, graphColor }) => (
  <div className={classes.circleContainer}>
    <ul>
      <li><DataCircle type={currentText.humidity} unit="%" data={formatNumber(currently.humidity * 100)} additionalData={null} graphColor={graphColor} /></li>
      <li><DataCircle type={currentText.wind} unit="m/s" data={formatNumber(currently.windSpeed)} additionalData={currently.windBearing} graphColor={graphColor} /></li>
      <li><DataCircle type={currentText.precipProb} unit="%" data={formatNumber(currently.precipProbability * 100)} additionalData={null} graphColor={graphColor} /></li>
      <li><DataCircle type={currentText.cloudCov} unit="%" data={formatNumber(currently.cloudCover * 100)} additionalData={null} graphColor={graphColor} /></li>
    </ul>
  </div>
)

export default CircleContainer
