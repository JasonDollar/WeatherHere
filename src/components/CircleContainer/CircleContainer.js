import React from 'react'
import PropTypes from 'prop-types'
import DataCircle from './DataCircle/DataCircle'
import { formatNumber } from '../../data/utils'
import { unitsNames } from '../../data/unitsNames'

import classes from './CircleContainer.module.scss'


const CircleContainer = ({
  currently, currentText, units, 
}) => (
  <div className={classes.circleContainer}>
    <ul>
      <li><DataCircle type={currentText.humidity} unit="%" data={formatNumber(currently.humidity * 100)} text={currentText} /></li>
      <li><DataCircle type={currentText.wind} unit={unitsNames[units].wind} data={formatNumber(currently.windSpeed)} additionalData={currently.windBearing} text={currentText} /></li>
      <li><DataCircle type={currentText.precipProb} unit="%" data={formatNumber(currently.precipProbability * 100)} text={currentText} /></li>
      <li><DataCircle type={currentText.cloudCov} unit="%" data={formatNumber(currently.cloudCover * 100)} text={currentText} /></li>
    </ul>
  </div>
)

export default CircleContainer

CircleContainer.propTypes = {
  currently: PropTypes.shape({
    apparentTemperature: PropTypes.number,
    cloudCover: PropTypes.number,
    dewPoint: PropTypes.number,
    humidity: PropTypes.number,
    icon: PropTypes.string,
    ozone: PropTypes.number,
    precipIntensity: PropTypes.number,
    precipProbability: PropTypes.number,
    pressure: PropTypes.number,
    summary: PropTypes.string,
    temperature: PropTypes.number,
    time: PropTypes.number,
    uvIndex: PropTypes.number,
    visibility: PropTypes.number,
    windBearing: PropTypes.number,
    windGust: PropTypes.number,
    windSpeed: PropTypes.number,
  }).isRequired,

  currentText: PropTypes.shape({
    today: PropTypes.string,
    humidity: PropTypes.string,
    wind: PropTypes.string,
    precipProb: PropTypes.string,
    precipType: PropTypes.string,
    precipInt: PropTypes.string,
    appTemp: PropTypes.string,
    dewPoint: PropTypes.string,
    pressure: PropTypes.string,
    cloudCov: PropTypes.string,
    visibility: PropTypes.string,
    uvIndex: PropTypes.string,
    sunrise: PropTypes.string,
    sunset: PropTypes.string,
    moonphase: PropTypes.string,
    moon: PropTypes.arrayOf(PropTypes.string),
    ozone: PropTypes.string,
    windGust: PropTypes.string,
    location: PropTypes.string,
    icon: PropTypes.string,
  }).isRequired,
  units: PropTypes.string.isRequired,
}