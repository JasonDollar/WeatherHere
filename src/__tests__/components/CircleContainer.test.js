import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { appTestData as data } from '../../data/fixtures'
import CircleContainer from '../../components/CircleContainer/CircleContainer' 

let currentText = data.text.pl.current
let currently = data.weather.currently
let units = data.units.si.id
let wrapper

beforeAll(() => {
  wrapper = shallow(
    <CircleContainer 
      currentText={currentText}
      currently={currently}
      units={units}
    />,
  )
})

test('CircleContainer matches Snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot()
})