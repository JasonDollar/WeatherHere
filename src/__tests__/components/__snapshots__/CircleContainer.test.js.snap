// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleContainer matches Snapshot 1`] = `
<div
  className="circleContainer"
>
  <ul>
    <li>
      <DataCircle
        additionalData={null}
        data={94}
        text={
          Object {
            "appTemp": "Temperatura odczuwalna",
            "cloudCov": "Zachmurzenie",
            "date": "Data",
            "dewPoint": "Punkt rosy",
            "humidity": "Wilgotność",
            "icon": "Ikona pogody",
            "location": "Nieznana lokacja",
            "moon": Array [
              "Nów",
              "Wschodzący sierp",
              "Pierwsza kwadra",
              "Wschodzący księżyc",
              "Pełnia",
              "Ubywający księżyc",
              "Trzecia kwadra",
              "Ubywający sierp",
            ],
            "moonphase": "Faza księżyca",
            "ozone": "Ozon",
            "precipInt": "Intensywność opadu",
            "precipProb": "Opady",
            "precipType": "Rodzaj opadu",
            "pressure": "Ciśnienie",
            "sun": "Słońce",
            "sunrise": "Wschód słońca",
            "sunset": "Zachód słońca",
            "today": "Dziś",
            "uvIndex": "Indeks UV",
            "visibility": "Widoczność",
            "wind": "Wiatr",
            "windGust": "Wiatr w porywach",
          }
        }
        type="Wilgotność"
        unit="%"
      />
    </li>
    <li>
      <DataCircle
        additionalData={99}
        data={5.7}
        text={
          Object {
            "appTemp": "Temperatura odczuwalna",
            "cloudCov": "Zachmurzenie",
            "date": "Data",
            "dewPoint": "Punkt rosy",
            "humidity": "Wilgotność",
            "icon": "Ikona pogody",
            "location": "Nieznana lokacja",
            "moon": Array [
              "Nów",
              "Wschodzący sierp",
              "Pierwsza kwadra",
              "Wschodzący księżyc",
              "Pełnia",
              "Ubywający księżyc",
              "Trzecia kwadra",
              "Ubywający sierp",
            ],
            "moonphase": "Faza księżyca",
            "ozone": "Ozon",
            "precipInt": "Intensywność opadu",
            "precipProb": "Opady",
            "precipType": "Rodzaj opadu",
            "pressure": "Ciśnienie",
            "sun": "Słońce",
            "sunrise": "Wschód słońca",
            "sunset": "Zachód słońca",
            "today": "Dziś",
            "uvIndex": "Indeks UV",
            "visibility": "Widoczność",
            "wind": "Wiatr",
            "windGust": "Wiatr w porywach",
          }
        }
        type="Wiatr"
        unit="m/s"
      />
    </li>
    <li>
      <DataCircle
        additionalData={null}
        data={1}
        text={
          Object {
            "appTemp": "Temperatura odczuwalna",
            "cloudCov": "Zachmurzenie",
            "date": "Data",
            "dewPoint": "Punkt rosy",
            "humidity": "Wilgotność",
            "icon": "Ikona pogody",
            "location": "Nieznana lokacja",
            "moon": Array [
              "Nów",
              "Wschodzący sierp",
              "Pierwsza kwadra",
              "Wschodzący księżyc",
              "Pełnia",
              "Ubywający księżyc",
              "Trzecia kwadra",
              "Ubywający sierp",
            ],
            "moonphase": "Faza księżyca",
            "ozone": "Ozon",
            "precipInt": "Intensywność opadu",
            "precipProb": "Opady",
            "precipType": "Rodzaj opadu",
            "pressure": "Ciśnienie",
            "sun": "Słońce",
            "sunrise": "Wschód słońca",
            "sunset": "Zachód słońca",
            "today": "Dziś",
            "uvIndex": "Indeks UV",
            "visibility": "Widoczność",
            "wind": "Wiatr",
            "windGust": "Wiatr w porywach",
          }
        }
        type="Opady"
        unit="%"
      />
    </li>
    <li>
      <DataCircle
        additionalData={null}
        data={92}
        text={
          Object {
            "appTemp": "Temperatura odczuwalna",
            "cloudCov": "Zachmurzenie",
            "date": "Data",
            "dewPoint": "Punkt rosy",
            "humidity": "Wilgotność",
            "icon": "Ikona pogody",
            "location": "Nieznana lokacja",
            "moon": Array [
              "Nów",
              "Wschodzący sierp",
              "Pierwsza kwadra",
              "Wschodzący księżyc",
              "Pełnia",
              "Ubywający księżyc",
              "Trzecia kwadra",
              "Ubywający sierp",
            ],
            "moonphase": "Faza księżyca",
            "ozone": "Ozon",
            "precipInt": "Intensywność opadu",
            "precipProb": "Opady",
            "precipType": "Rodzaj opadu",
            "pressure": "Ciśnienie",
            "sun": "Słońce",
            "sunrise": "Wschód słońca",
            "sunset": "Zachód słońca",
            "today": "Dziś",
            "uvIndex": "Indeks UV",
            "visibility": "Widoczność",
            "wind": "Wiatr",
            "windGust": "Wiatr w porywach",
          }
        }
        type="Zachmurzenie"
        unit="%"
      />
    </li>
  </ul>
</div>
`;
